core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/BlockContext.scala

```diff
@@ -36,15 +36,20 @@ sealed trait BlockContext {
 
   def responseTransformations: List[ResponseTransformation]
 
-  def isCurrentGroupEligible(permittedGroupIds: PermittedGroupIds): Boolean = {
-    userMetadata.currentGroupId match {
-      case Some(preferredGroupId) =>
-        requestContext.uriPath.isCurrentUserMetadataPath || permittedGroupIds.matches(preferredGroupId)
-      case None =>
+  def isCurrentGroupEligible(groupsPotentiallyPermittedByRule: GroupsPotentiallyPermittedByRule): Boolean = {
```

Comment:
I see some problem here. ROR request could have a header with the current group (the current group is related to the picked value in the tenancy selector value at the ROR KBN side). 

Eg. `GET /_search/index1 -H "x-ror-current-group: group1"`

During block evaluation (the one with an authorization rule in it), ROR determines the allowed groups list (obviously, allowed groups can have not only full names groups, but also groups with wildcards). 
The `isCurrentGroupEligible` method simply checks if the current group matches the allowed groups. 

Currently, we deal here with potentially permitted groups. But we should deal with the already resolved and truly permitted groups. Maybe it's about the naming but seeing "All" allowed groups in this place, I'm pretty sure, sth is wrong here.